p2p/pex/addrbook.go

```diff
@@ -93,13 +96,14 @@ type addrBook struct {
 
 // NewAddrBook creates a new address book.
 // Use Start to begin processing asynchronous address updates.
-func NewAddrBook(filePath string, routabilityStrict bool) *addrBook {
+func NewAddrBook(cfg *config.P2PConfig) *addrBook {
```

Comment:
Should we pass the whole config here or rather create a 

```
AddrBookConfig struct {
  Filepath string
  Strict bool
  NeedAddressThreshold int
}
```

P2P config is gonna grow over time and it might become inconvenient to create it every time say you need a book. Plus addrbook now depends on `config` package 👎 

I feel like 
```
NewAddrBook(filename string, config *AddrBookConfig) {
}
```
is a better looking API. Thoughts?

Reply:
In general it seems undesirable to have entities holding handles to (mutable) configuration structs. It leads to leakage of environmental concerns into business logic, e.g.: having a test related struct field checked in the middle of an algorithm. It leads to lazy implemenations and non-solid boundaries.

Instead APIs should express dependencies, implementations should come with sane defaults and configuration can be handled with options demonstrated in other parts of the codebase.